sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Represents a range of bytes within an HTTP resource.
+ * <p>
+ * The range starts at the {@link #getOffset() offset} inclusively and ends at {@link #getOffset() offset} + {@link
+ * #getLength() legnth} exclusively, or offset + length - 1.
+ * <p>
+ * If {@link #getLength() length} is unspecified, null, then the range extends to the end of the HTTP resource.
+ */
+public final class HttpRange {
```

Comment:
This seems to be generic enough to be protocol agnostic. Should we have `Http` in the name?

Reply:
While the functionality seems generic, the toString returning `bytes=<offset>-<offset + length>` is tightly tied to what the `Range` header expected in HTTP.

Reply:
Does `toString()` have to format it in a way that it is tied to Http?

Reply:
Having `toString` be `HTTP` specific isn't a strong requirement. The class could be made into a more generic `Range` and utility methods could be provided to format the range into an HTTP `Range` header.

Reply:
Let's leave it as HttpRange for now as there's no other usecase that we know of that required `Range` type.